### ISSUE TITLE
Chakra Storybook: TypeError "theme.background is undefined" (TypeScript / React)

### ISSUE THREAD

**Issue by TylerAPfledderer** [chakra-ui/chakra-ui#5515]
### Description

I recreated this issue in a fresh project, following steps to install the `create-react-app` template for Chakra and Typescript per docs, and the Chakra addon and `@chakra-ui/icons` also per Chakra docs along with initializing Storybook. Upon booting storybook, it appears that you able to see a component no issue, but when navigating to the docs tabs, The error 'theme.background is undefined' apppears, I switch back to the visual, and the error persists. In addition to the example stories that Storybook provides, I made a test component that uses a Chakra component and getting the same result.

### Link to Reproduction

https://codesandbox.io/s/musing-monad-mrzmg

### Steps to reproduce

A fresh project.
1. Run `yarn create react-app my-app --template @chakra-ui/typescript`
2. Run `npx sb init`
3. Run ` yarn add -D @chakra-ui/storybook-addon`
and `yarn add @chakra-ui/icons` per [Chakra docs on Storybook](https://chakra-ui.com/guides/integrations/with-storybook)
4. Added `@chakra-ui/storybook-addon` to `.storybook/main.js` per docs
5. Created a test component using two Chakra Components, and created a default story for this component.
6. Run `yarn storybook`
7. View a component (Storybook example or `TestComponent`)
8. Select the `Docs` tab if error did not appear immediately
9. Switch back to `Canvas` to see error persistence

No other configurations done.

### Chakra UI Version

1.8.1

### Browser

Firefox 96.0.3

### Operating System

- [ ] macOS
- [X] Windows
- [ ] Linux

### Additional Information

@chakra-ui/storybook-addon `v1.0.1`

Is there supposed to be an extra step to satisfy TypeScript?

Screenshot:
![screenshot](https://user-images.githubusercontent.com/65234762/152372614-fb79d782-f5cd-4a9e-964b-4466f517a5dd.png)

**Comment by TylerAPfledderer** [comment]
Hey @segunadebayo!

I tried it, and ended up with the following console error:
```bash
$ yarn story
yarn run v1.22.5
$ start-storybook -p 6006 -s public
info @storybook/react v6.4.19
info
(node:6312) DeprecationWarning: --static-dir CLI flag is deprecated, see:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated---static-dir-cli-flag
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Loading presets
info => Serving static files from ./public at /
info => Loading Webpack configuration from `node_modules\react-scripts`
info => Removing existing JavaScript and TypeScript rules.
info => Modifying Create React App rules.
info => Using default Webpack4 setup
ERR! WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
ERR!  - configuration.module.rules[5].oneOf[9].type should be one of these:
ERR!    "javascript/auto" | "javascript/dynamic" | "javascript/esm" | "json" | "webassembly/experimental"
ERR!    -> Module type to use for the module
ERR!     at webpack (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\builder-webpack4\node_modules\webpack\lib\webpack.js:31:9)
ERR!     at Object.start (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\builder-webpack4\dist\cjs\index.js:92:18)
ERR!     at async Promise.all (index 0)
ERR!     at async storybookDevServer (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\dev-server.js:126:28)
ERR!     at async buildDevStandalone (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\build-dev.js:115:31)
ERR!     at async buildDev (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\build-dev.js:161:5)
ERR!  WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
ERR!  - configuration.module.rules[5].oneOf[9].type should be one of these:
ERR!    "javascript/auto" | "javascript/dynamic" | "javascript/esm" | "json" | "webassembly/experimental"
ERR!    -> Module type to use for the module
ERR!     at webpack (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\builder-webpack4\node_modules\webpack\lib\webpack.js:31:9)
ERR!     at Object.start (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\builder-webpack4\dist\cjs\index.js:92:18)
ERR!     at async Promise.all (index 0)
ERR!     at async storybookDevServer (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\dev-server.js:126:28)
ERR!     at async buildDevStandalone (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\build-dev.js:115:31)
ERR!     at async buildDev (C:\Users\tyler\Documents\Sites\lesson-scheduler\node_modules\@storybook\core-server\dist\cjs\build-dev.js:161:5)
```

**Comment by TylerAPfledderer** [comment]
I also made a fresh "test" project using the steps in the description of this issue, adding the suggested solution, and received this same result.

**Comment by srt4rulez** [comment]
I'm also having the same issue. Here's a screenshot of my error, but it only shows in my console: 

![image](https://user-images.githubusercontent.com/4718542/160296605-2e0d2e94-9200-4c5e-b659-6e1d6a8937ca.png)

```
14:57:42.424 You are loading @emotion/react when it is already loaded. Running multiple instances may cause problems. This can happen if multiple versions are used, or if multiple builds of the same version are used. [emotion-react.browser.cjs.js:377](webpack://<redacted org>/react-component-library/node_modules/@emotion/react/dist/emotion-react.browser.cjs.js)
14:57:42.451
Uncaught TypeError: can't access property "content", theme.background is undefined
    node_modules vendors-node_modules_storybook_addon-docs_dist_esm_blocks_index_js.iframe.bundle.js:8820
    handleInterpolation emotion-serialize.browser.esm.js:137
    serializeStyles emotion-serialize.browser.esm.js:251
    Styled emotion-styled-base.browser.cjs.js:160
    withEmotionCache emotion-element-cbed451f.browser.esm.js:36
[DocsPage.js:45](webpack://<redacted org>/react-component-library/node_modules/@storybook/components/dist/esm/blocks/DocsPage.js)
14:57:42.453
XHRPOSThttp://localhost:6006/runtime-error
[[HTTP/1.1 404 Not Found 3ms]]()

14:57:42.454
The above error occurred in the <Styled(div)> component:

withEmotionCache/<@http://localhost:6006/vendors-node_modules_babel_runtime_regenerator_index_js-node_modules_bem-react_classname_inde-0c2a2d.iframe.bundle.js:37981:66
MDXProvider@http://localhost:6006/vendors-node_modules_storybook_addon-docs_dist_esm_blocks_index_js.iframe.bundle.js:2798:23
ThemeProvider@http://localhost:6006/vendors-node_modules_storybook_addon-docs_dist_esm_blocks_index_js.iframe.bundle.js:874:21
SourceContainer@http://localhost:6006/vendors-node_modules_storybook_addon-docs_dist_esm_blocks_index_js.iframe.bundle.js:4302:18
DocsContainer@http://localhost:6006/vendors-node_modules_storybook_addon-docs_dist_esm_blocks_index_js.iframe.bundle.js:3609:17

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries. [react-dom.development.js:20085](webpack://<redacted org>/react-component-library/node_modules/react-dom/cjs/react-dom.development.js)
```

I tried https://github.com/chakra-ui/chakra-ui/issues/5450#issuecomment-1021715710 but I don't get any compile errors, just the same console error. I am running [webpack5 ](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#webpack-5) though for storybook. 

I'm guessing the warning at the top has something to do with this, running multiple versions of `@emotion/react`. 

Here's the package versions I'm running:

- @chakra-ui/react@1.8.6
- @chakra-ui/storybook-addon@1.0.3
- storybook 6.4.19

I'm not using CRA, I have my own webpack setup.

**Comment by srt4rulez** [comment]
I fixed it for me by adding the alias' from https://github.com/storybookjs/storybook/issues/16716#issuecomment-1020613602

It seems that https://github.com/storybookjs/storybook/pull/17000 is suppose to fix the issue without a workaround, which is to be released with storybook v6.5.0.
